HTTPServerProject.Tests/HTTPServerReadTests/TestReadStreams.cs

```diff
@@ -0,0 +1,104 @@
+using HTTPServerRead.Streams;
+
+namespace HTTPServerReadTests.Streams;
+
+public class UnitTestsForReadStreams
+    {
+        [Fact]
+        public void ReaderIsClosedTest()
+        {
+            var stream = new List<string>();
+            var reader = new TestReadStreams(stream);
+            reader.Close();
+            var readerHasClosed = reader.closeCalled;
+
+            Assert.True(readerHasClosed);
+        }    
+    }
+
+public class TestReadStreams : IReadStreams
+{
+    List<string> sArr = new List<string>();
+	public bool closeCalled = false;	
+
+    public TestReadStreams(List<string> arr)
+    {
+        sArr = arr;
+    }
+
+    public string ReadLine()
+    {
+        if (sArr.Count > 0)
+        {
+            var input = sArr[0];
+            sArr.RemoveAt(0);
+            return input;
+        }
+
+        return "";
+    }
+
+    public int Read()
+    {
+        var input = sArr[0];
+        if (input == "")
+        {
+            return -1;
+        }
+        else
+        {
+            var c = input[0];
+            var ascii_c = (int)c;
+
+            if (input.Length > 1)
+            {
+                sArr[0] = input.Substring(1);
+            }
+            else
+            {
+                sArr[0] = "";
```

Comment:
What if the input is only one character, won't you lose the character?

Reply:
No, because the one character would assigned to `var c` before the `if/else` statement.